### PR TITLE
Adds typescript support

### PR DESCRIPTION
**Pull request by mrclay**
TODO

- [x] get TS to stop complaining about webpack externals
- [x] bring back `import Drupal from 'drupal'` and friends
- [x] test tsx files
- [x] test mixing js/ts files in same script
- [x] address some [Codacy issues](https://app.codacy.com/gh/forumone/gesso/pullRequest?bid=27521070&prid=8821041)

### CODE REVIEW

**Comment by kmonahan** [comment]
I think this is covered by your last to-do, but want to call out just in case that we need to test that TS is supported but not required -- even if the base JS files are all converted to TS, new scripts need to be able to be added as JS without types.

File: source/@types/drupal/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  const Drupal: {
+    attachBehaviors(context?: HTMLElement | Document, settings?: Settings): void;
+
+    throwError(error: Error | string): void;
+
+    checkPlain(str: string): string;
+
+    formatString(str: string, args: object): string;
+
+    stringReplace(str: string, args: object, keys?: any): string;
```
Comment (slucero) [inline source/@types/drupal/index.d.ts:30]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/forumone/gesso/pullRequest?prid=8821041)
```suggestion
    stringReplace(str: string, args: object, keys?: unknown): string;
```

File: source/@types/drupal/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      (func: string, options: any): string | HTMLElement;
+
+      placeholder(str: string): string;
+
+      [key: string]: (...args: any[]) => string | HTMLElement;
+    };
+
+    behaviors: Record<string, Behavior>;
+
+    [key: string]: any;
```
Comment (slucero) [inline source/@types/drupal/index.d.ts:62]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/forumone/gesso/pullRequest?prid=8821041)
```suggestion
    [key: string]: unknown;
```

File: source/@types/drupal/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+      isLocal(url: string): boolean;
+    };
+
+    theme: {
+      (func: string, options: any): string | HTMLElement;
+
+      placeholder(str: string): string;
+
+      [key: string]: (...args: any[]) => string | HTMLElement;
```
Comment (slucero) [inline source/@types/drupal/index.d.ts:57]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/forumone/gesso/pullRequest?prid=8821041)
```suggestion
      [key: string]: (...args: unknown[]) => string | HTMLElement;
```

File: source/@types/drupal/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    url: {
+      (path: string): string;
+
+      toAbsolute(url: string): string;
+
+      isLocal(url: string): boolean;
+    };
+
+    theme: {
+      (func: string, options: any): string | HTMLElement;
```
Comment (slucero) [inline source/@types/drupal/index.d.ts:53]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/forumone/gesso/pullRequest?prid=8821041)
```suggestion
      (func: string, options: unknown): string | HTMLElement;
```

File: source/@types/drupal/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+    throwError(error: Error | string): void;
+
+    checkPlain(str: string): string;
+
+    formatString(str: string, args: object): string;
+
+    stringReplace(str: string, args: object, keys?: any): string;
+
+    t(str: string, args?: any, options?: string): string;
```
Comment (slucero) [inline source/@types/drupal/index.d.ts:32]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/forumone/gesso/pullRequest?prid=8821041)
```suggestion
    t(str: string, args?: unknown, options?: string): string;
```

File: source/06-utility/build-test/Component-jsx.jsx [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import moduleTS from './_module-ts';
+
+function ComponentJsx() {
+  return <div>{moduleTS()}</div>;
+};
```
Comment (slucero) [inline source/06-utility/build-test/Component-jsx.jsx:6]: Codacy has a fix for the issue: [Unnecessary semicolon.](https://app.codacy.com/gh/forumone/gesso/pullRequest?prid=8821041)
```suggestion
}
```

Comment (slucero) [inline source/06-utility/build-test/Component-jsx.jsx:6]: Codacy has a fix for the issue: [Unnecessary semicolon.](https://app.codacy.com/gh/forumone/gesso/pullRequest?prid=8821041)
```suggestion
}
```

**Review by kmonahan — APPROVED** [review]
TypeScript support! 🎉 @coreylafferty This one's ready to merge.

**Review by kmonahan — APPROVED** [review]

**Pull request merged**